src/Components/authenticate/providers/UserProvider.jsx

```diff
@@ -1,15 +1,31 @@
 import React, { Component, createContext } from "react";
 import { auth } from "../../../firebase";
-
 export const UserContext = createContext({ user: null });
+
+
 class UserProvider extends Component {
   state = {
     user: null
   };
 
   componentDidMount = () => {
     auth.onAuthStateChanged(userAuth => {
-      this.setState({user: userAuth});
+      // The event stuff here is only done because when sign in is triggered on the same page
+      // it won't cause a rerender.
```

Comment:
lol this wording is confusing idk what you are trying to say, is it just that using an event will prevent a page refresh? can you reword

Reply:
I'll add this in a comment but what I mean is what I'm saying here normally if the authentication status is changed it won't cause a rerender, only if it's on a different page. That's why I'm adding the dispatch event to force trigger.